backend/main.py

```diff
@@ -268,6 +271,54 @@ def update_leader_board_total(session: Session) -> None:
         logger.exception("Error during user states update (total)")
 
 
+@app.on_event("startup")
+def set_startup_time() -> None:
+    global startup_time
+    startup_time = utcnow()
+
+
+def update_all_user_streak(session: Session) -> None:
+    statement = select(User)
+    result = session.exec(statement).all()
+    return result
+
+
+@app.on_event("startup")
+@repeat_every(seconds=60 * settings.USER_STREAK_UPDATE_INTERVAL, wait_first=False)
+@managed_tx_function(auto_commit=CommitMode.COMMIT)
+def update_user_streak(session: Session) -> None:
+    try:
+        global startup_time
+        current_time = utcnow()
+        timedelta = current_time - startup_time
+        result = update_all_user_streak(session=session)
+        if timedelta.days >= 0:
+            # Update only greater than 24 hours . Do nothing
+            logger.debug("Process timedelta greater than 24h")
+            if result is not None:
+                for user in result:
+                    logger.debug(f"{type(user)}... {user}")
+                    last_activity_date = user.last_activity_date
+                    streak_last_day_date = user.streak_last_day_date
+                    current_time_no_tz = current_time.replace(tzinfo=None)
+                    logger.debug(f"{current_time_no_tz}: {last_activity_date}, {streak_last_day_date}")
+                    if last_activity_date is not None:
+                        lastactitvitydelta = current_time_no_tz - last_activity_date
+                        if lastactitvitydelta.days > 1 or user.streak_days is None:
+                            user.streak_days = 0
```

Comment:
we need to set `user.streak_last_day_date` to current time here too, don't we?

Reply:
yes, it is needed. I added it.